marimo/_server/api/interrupt.py

```diff
@@ -55,18 +55,28 @@ def _interrupt_handler(self) -> None:
             self.shutdown()
 
         try:
-            if GLOBAL_SETTINGS.YES:
-                self.shutdown()
-                return
+            try:
+                if GLOBAL_SETTINGS.YES:
+                    self.shutdown()
+                    return
 
-            response = input(
-                f"\r{TAB}\033[1;32mAre you sure you want to quit?\033[0m "
-                "\033[1m(y/n)\033[0m: "
-            )
-            if response.lower().strip() == "y":
+                response = input(
+                    f"\r{TAB}Are you sure you want to quit? (y/n): "
+                )
+                self._time_of_last_confirmation = time.time()
+                if response.lower().strip() == "y":
+                    self.shutdown()
+                    return
+            except (KeyboardInterrupt, EOFError, asyncio.CancelledError):
+                print_()
                 self.shutdown()
                 return
-        except (KeyboardInterrupt, EOFError, asyncio.CancelledError):
+        except KeyboardInterrupt:
+            # This is a hack to workaround the fact that uv can send two SIGINT for
```

Comment:
Thought through this 

```python
def handle(depth=0):
    try:
        if depth < MAX_ATTEMPTS:
            return handle(depth + 1)
        ...
    except:
        ...
```

but **still** a chance for a race condition killing it before recursion.

I also considered it might be best to debounce in the sandbox pass to the process
BUT another hypothesis for the double sigint, is that hitting ctrl-c in your terminal sends sigint to the process group (or maybe sending 1 to uv and uv is sending to the whole process group) so manually sending the sig int in sandbox at all is actually duplicate behavior.

If it's not duplicate behavior (or maybe terminal / uv version specific), then I think this is good enough

Reply:
Yea I was just thinking about trying something like this. Will try it out

Reply:
The issue is that `input()` is not reentrant so we can't re-enter `_input_handler.` I tried some other logic, to have a custom handler that raises the KeyboardInterrupt, keeping track of how many times it was called. But that also has a race and ended up being more buggy. Kept the logic as is.